docker-compose.test.yml

```diff
@@ -0,0 +1,15 @@
+version: "3.8"
+services:
+  tests:
+    build:
+      context: .
+      target: ci
+      dockerfile: Dockerfile.test
+    environment:
+      - "MONGO_TEST_BASE_URI=mongodb://db:27017"
+    depends_on:
+      - db
+  db:
```

Comment:
Potentially nicer to use a separate name for this service compared to the normal compose file to avoid conflicts between the two. But then again, you do use a named "test" database for the tests so it shouldn't be an issue.